redash/handlers/admin.py

```diff
@@ -32,9 +33,14 @@ def outdated_queries():
 @require_super_admin
 @login_required
 def queries_tasks():
-    waiting = QueryTaskTracker.all(QueryTaskTracker.WAITING_LIST)
-    in_progress = QueryTaskTracker.all(QueryTaskTracker.IN_PROGRESS_LIST)
-    done = QueryTaskTracker.all(QueryTaskTracker.DONE_LIST, limit=50)
+    global_limit = int(request.args.get('limit')) if request.args.get('limit') is not None else 50
+    waiting_limit = int(request.args.get('waiting_limit')) if request.args.get('waiting_limit') is not None else None
+    progress_limit = int(request.args.get('progress_limit')) if request.args.get('progress_limit') is not None else None
+    done_limit = int(request.args.get('done_limit')) if request.args.get('done_limit') is not None else None
```

Comment:
`{}.get` can take a fallback value if the key isn't present, so I think you can simplify this with:

```python
global_limit = int(request.args.get('limit', 50))
waiting_limit = int(request.args.get('waiting_limit', global_limit))
progress_limit = int(request.args.get('progress_limit', global_limit))
done_limit = int(request.args.get('done_limit', global_limit))
```

And then you can pass the different limits as is without the `x or global_limit` part.